cmd/podman/system/connection/add.go

```diff
@@ -141,6 +141,19 @@ func add(cmd *cobra.Command, args []string) error {
 
 	switch uri.Scheme {
 	case "ssh":
+		// ensure the Identity provided is a valid file
+		info, err := os.Stat(entities.Identity)
+		switch {
+		case errors.Is(err, os.ErrNotExist):
+			logrus.Warnf("%q does not exist", entities.Identity)
+		case errors.Is(err, os.ErrPermission):
+			logrus.Warnf("You do not have permission to read %q", entities.Identity)
```

Comment:
Why print a warning? Your commit says prevent adding but this doesn't prevent it at all.
IMO simply returning the error is the easiest and there is no need for us to special case each error, the orginal error should have enough information.